module/icp/algs/aes/aes_impl.c

```diff
@@ -951,15 +954,18 @@ rijndael_key_setup_enc_raw(uint32_t rk[], const uint32_t cipherKey[],
 static void
 aes_setupkeys(aes_key_t *key, const uint32_t *keyarr32, int keybits)
 {
-	if (intel_aes_instructions_present()) {
+#if defined(HAVE_AES)
+	if (zfs_aes_available()) {
 		key->flags = INTEL_AES_NI_CAPABLE;
```

Comment:
Should this be changed to use the function pointer model that the vdev raidz code uses?

Reply:
What's the test situation like here? If the implementations are selectable, then it's possible to run the tests against the various implementations (e.g. with and without AES-NI).

Reply:
That'd be a nice addition to facilitate testing and to add visibility to make it easy to verify you are using the accelerated version.  It also nicely resolves the issue of caching the result, so I think it would be a nice addition.